src/plenigo/models/Order.php

```diff
@@ -0,0 +1,322 @@
+     */
+    public function setVatNumber($vatNumber) {
+        $this->vatNumber = $vatNumber;
+    }
+
+    /**
+     * Setter method
+     * @param number $shippingCosts
+     */
+    public function setShippingCosts($shippingCosts) {
+        $this->shippingCosts = $shippingCosts;
+    }
+
+    /**
+     * Setter method
+     * @param number $shippingCostsTaxes
+     */
+    public function setShippingCostsTaxes($shippingCostsTaxes) {
+        $this->shippingCostsTaxes = $shippingCostsTaxes;
+    }
+
+    /**
+     * Setter method
+     * @param number $purchaseOrderIndicator
+     */
+    public function setPurchaseOrderIndicator($purchaseOrderIndicator) {
+        $this->purchaseOrderIndicator = $purchaseOrderIndicator;
+    }
+
+    /**
+     * Setter method
+     * @param number $discount
+     */
+    public function setDiscount($discount) {
+        $this->discount = $discount;
+    }
+
+    /**
+     * Setter method
+     * @param int $discountPercentage
+     */
+    public function setDiscountPercentage($discountPercentage) {
+        $this->discountPercentage = $discountPercentage;
+    }
+
+    /**
+     * Setter method
+     * @param array $orderItems
+     */
+    public function setOrderItems($orderItems) {
+        $this->orderItems = $orderItems;
+    }
+
+    /**
+     * Creates a Order instance from an array map.
+     *
+     * @param array $map The array map to use for the instance creation.
+     * @return Order instance.
+     */
+    public static function createFromMap($map) {
+        $instance = new Order();
+        $instance->setOrderId(isset($map['orderId']) ? $map['orderId'] : null);
+        $instance->setOrderDate(isset($map['orderDate']) ? $map['orderDate'] : null);
+        $instance->setCustomerId(isset($map['customerId']) ? $map['customerId'] : null);
+        $instance->setCurrency(isset($map['currency']) ? $map['currency'] : null);
+        $instance->setCumulatedPrice(isset($map['cumulatedPrice']) ? $map['cumulatedPrice'] : null);
+        $instance->setVatNumber(isset($map['vatNumber']) ? $map['vatNumber'] : null);
+        $instance->setShippingCosts(isset($map['shippingCosts']) ? $map['shippingCosts'] : null);
+        $instance->setShippingCostsTaxes(isset($map['shippingCostsTaxes']) ? $map['shippingCostsTaxes'] : null);
+        $instance->setPurchaseOrderIndicator(isset($map['purchaseOrderIndicator']) ? $map['purchaseOrderIndicator'] : null);
+        $instance->setDiscount(isset($map['discount']) ? $map['discount'] : null);
+        $instance->setDiscountPercentage(isset($map['discountPercentage']) ? $map['discountPercentage'] : null);
+        $instance->setOrderItems(isset($map['orderItems']) ? $map['orderItems'] : null);
+        if (isset($map['orderItems']) && !is_null($map['orderItems']) && count(($map['orderItems'])) > 0) {
+            $arrItems = array();
+            foreach ($map['orderItems'] as $oItem) {
+                $arrItems[] = OrderItem::createFromMap((array) $oItem);
```

Comment:
This solution is equivalent without creating another variable ($item in your example)